src/core/memory.d

```diff
@@ -708,6 +712,15 @@ struct GC
         return gc_stats();
     }
 
+    /**
+     * Resets runtime stats for currently active GC implementation of current thread
+     * See `core.memory.GC.Stats` for list of available metrics.
+     */
+    static void resetThreadLocalStats() nothrow
```

Comment:
A reset function is never a good idea with independent clients checking the same value. Just use difference to a previous value.

Reply:
The problem with this is detecting overflows of the counter for long lived processes. But it might be acceptable anyway. I have the feeling there were other reasons for the reset, but I can't remember right now.

Reply:
The counter will overflow just the same as the difference between two consecutive calls.

Reply:
One advantage for applications is they can reset the counter and then just look at it later. If they can't reset, they need to save the old value then subtract it from the current value.

Reply:
It's still a bad API especially with multiple clients. Do you also reset the system clock if you are doing benchmarks?

Reply:
@rainers has a point. The overflowing is still passing quite some complexity to the client but it's true there are workarounds to detect it and deal with it.